src/cargo/ops/cargo_clean.rs

```diff
@@ -21,17 +21,24 @@ pub fn clean(manifest_path: &Path, opts: &CleanOptions) -> CargoResult<()> {
 
     // If we have a spec, then we need to delete some packages, otherwise, just
     // remove the whole target directory and be done with it!
+    //
+    // Note that we don't bother grabbing a lock here as we're just going to
+    // blow it all away anyway.
```

Comment:
Yeah this'll mean that they produce weird results most likely. This is somewhat difficult to lock as the target directory is somewhat granularly locked as opposed to "all at once". Figuring out all the locks we'd need to grab to lock the whole thing seemed more trouble than it was worth.

I also felt that in general we should do as much as we can to help reasonable concurrent requests of Cargo work out, but conflicting things like this seemed fine to me to cause some weird results.